MySolution.rb

```diff
@@ -0,0 +1,48 @@
+class MySolution
+
+    #problem1
+    def sumOfEven(array)
+        sum = 0
+        for i in array
+            if i.even?
+                sum += i
+            end
+        end
+        return sum
+    end
+    
+    #problem2
+    def getLargest2Numbers(array)
+        bigMax = -1
+        smallMax = -1
+        for i in array
+            if i > bigMax and i > smallMax #if bigger then both
+                smallMax = bigMax #new small is big
+                bigMax = i  #update big
+            else
+                if i > smallMax #bigger only then small?
+                    smallMax = i #uptade small
+                end
+            end
+        end
+        result = [smallMax, bigMax]
+        return result
+    end
+
+    #problem3
+    def hasDublicates(array)
+        #without temporary data structure to help, or sorting the problem
+        #goes to compare each element with each other
+        for i in 0..array.length * array.length 
```

Comment:
Very cool solution.

But you technically cheated `Try to find a solution by executing just one loop.` you masked a loop in a loop with your 2 indexes.

Reply:
Thanks. 
Yes I'm aware that is technically a cheat, as I wrote in the comment I couldn't see anyway how to compare each element with each other. I'll try to find a different approach